.travis.yml

```diff
@@ -1,35 +1,100 @@
 language: php
-sudo: required
-dist: trusty
 
-matrix:
+jobs:
   include:
-  - php: 7.0
-  - php: 7.1
-  - php: 7.2
-  - php: 5.5.38
-  - php: 5.6.25
-  - php: hhvm-3.24
-    group: edge
-  fast_finish: true
-
-before_script:
-  - pecl install grpc || echo 'Failed to install grpc'
-  - if [[ $TRAVIS_PHP_VERSION =~ ^7 ]]; then pecl install stackdriver_debugger-alpha || echo 'Failed to install stackdriver_debugger'; fi
-  - composer install
-  - if [[ $TRAVIS_PHP_VERSION =~ ^hhvm ]]; then composer --no-interaction --dev remove google/protobuf google/gax google/proto-client; fi
-  - dev/google-cloud credentials
+    - php: 7.2
+      install:
+        - pecl install grpc || echo 'Failed to install grpc'
+        - pecl install stackdriver_debugger-alpha || echo 'Failed to install stackdriver_debugger'
+        - composer self-update
+        - travis_retry composer install
+        - ./dev/google-cloud credentials
+
+    - php: 7.1
+      install:
+        - pecl install grpc || echo 'Failed to install grpc'
+        - pecl install stackdriver_debugger-alpha || echo 'Failed to install stackdriver_debugger'
+        - composer self-update
+        - travis_retry composer install
+        - ./dev/google-cloud credentials
+
+    - php: 7.0
+      install:
+        - pecl install grpc || echo 'Failed to install grpc'
+        - pecl install stackdriver_debugger-alpha || echo 'Failed to install stackdriver_debugger'
+        - composer self-update
+        - travis_retry composer install
+        - ./dev/google-cloud credentials
+
+    - php: 5.6.25
+      install:
+        - pecl install grpc || echo 'Failed to install grpc'
+        - composer self-update
+        - travis_retry composer install
+        - ./dev/google-cloud credentials
+
+    - php: 5.5.38
+      install:
+        - pecl install grpc || echo 'Failed to install grpc'
+        - composer self-update
+        - travis_retry composer install
+        - ./dev/google-cloud credentials
+
+    - php: hhvm-3.24
+      fast_finish: true
+      install:
+        - composer self-update
+        - travis_retry composer install
+        - ./dev/google-cloud credentials
+        - composer --no-interaction --dev remove google/protobuf google/gax google/proto-client
+
+    - stage: "Build Docs"
+      php: 7.2
+      install:
+      - travis_retry composer install
+      script:
+      - ./dev/sh/build-docs
+
+    - stage: "Push Docs"
+      php: 7.2
+      install:
+      - travis_retry composer install
+      script:
+      - ./dev/sh/build-docs
+      - ./dev/sh/push-docs $TRAVIS_BRANCH
+
+    - stage: "Split Components"
+      php: 7.2
+      install:
+      - travis_retry composer install
+      script:
+      - ./dev/sh/trigger-split
+      notifications:
+        email:
+          recipients:
```

Comment:
@chingor13 would you be like to be added to the list as well?

Reply:
Sure

Reply:
done!